python/plugins/processing/ui/resultsdockbase.ui

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
+ <class>DockWidget</class>
+ <widget class="QDockWidget" name="DockWidget">
```

Comment:
Can you make this a QgsDockWidget instead of a QDockWidget? QgsDockWidget has extra logic which would be handy in this use. Specifically if you use QgsDockWidget setUserVisible (https://qgis.org/api/classQgsDockWidget.html#aa5c1e81377a85c5ba4e0fa4a287dca46) then there's improved behaviour with raising hidden docks and logic when to close them. This is the same logic used by the style dock widget and nicely handles tabbed docks.